spinoso-time/src/time/tzrs/offset.rs

```diff
@@ -0,0 +1,273 @@
+        let offset_name = offset_hhmm_from_seconds(offset);
+        let local_time_type =
+            LocalTimeType::new(offset, false, Some(offset_name.as_bytes())).expect("Couldn't create fixed offset");
+        Self::Fixed([local_time_type])
+    }
+
+    /// Generate an offset based on a provided [`tz::timezone::TimeZoneRef`]
+    ///
+    /// This can be combined with [`tzdb`] to generate offsets based on predefined iana time zones
+    ///
+    /// ```
+    /// use spinoso_time::tzrs::Offset;
+    /// use tzdb::time_zone::pacific::AUCKLAND;
+    /// let offset = Offset::tz(AUCKLAND);
+    /// ```
+    ///
+    /// [`tz:timezone::TimeZoneRef`]: https://docs.rs/tz-rs/0.6.9/tz/timezone/struct.TimeZoneRef.html
+    /// [`tzdb`]: https://docs.rs/tzdb/latest/tzdb/index.html
+    #[inline]
+    #[must_use]
+    pub fn tz(tz: TimeZoneRef<'static>) -> Self {
+        Self::Tz(tz)
+    }
+
+    /// Returns a `TimeZoneRef` which can be used to generate and project _time_.
+    #[inline]
+    #[must_use]
+    pub fn time_zone_ref(&'a self) -> TimeZoneRef<'a> {
+        match self {
+            Self::Utc => TimeZoneRef::utc(),
+            Self::Fixed(local_time_types) => match TimeZoneRef::new(&[], local_time_types, &[], &None) {
+                Ok(tz) => tz,
+                Err(_) => GMT,
+            },
+
+            Self::Tz(zone) => *zone,
+        }
+    }
+}
+
+impl From<&str> for Offset {
+    /// Construct a Offset based on the [accepted MRI values]
+    ///
+    /// Accepts:
+    ///
+    /// - `[+/-]HH[:]MM`
+    /// - A-I representing +01:00 to +09:00
+    /// - K-M representing +10:00 to +12:00
+    /// - N-Y representing -01:00 to -12:00
+    /// - Z representing 0 offset
+    ///
+    /// [accepted MRI values]: https://ruby-doc.org/core-2.6.3/Time.html#method-c-new
+    #[inline]
+    #[must_use]
+    fn from(input: &str) -> Self {
+        match input {
+            "A" => Self::fixed(1),
+            "B" => Self::fixed(2),
+            "C" => Self::fixed(3),
+            "D" => Self::fixed(4),
+            "E" => Self::fixed(5),
+            "F" => Self::fixed(6),
+            "G" => Self::fixed(7),
+            "H" => Self::fixed(8),
+            "I" => Self::fixed(9),
+            "K" => Self::fixed(10),
+            "L" => Self::fixed(11),
+            "M" => Self::fixed(12),
+            "N" => Self::fixed(-1),
+            "O" => Self::fixed(-2),
+            "P" => Self::fixed(-3),
+            "Q" => Self::fixed(-4),
+            "R" => Self::fixed(-5),
+            "S" => Self::fixed(-6),
+            "T" => Self::fixed(-7),
+            "U" => Self::fixed(-8),
+            "V" => Self::fixed(-9),
+            "W" => Self::fixed(-10),
+            "X" => Self::fixed(-11),
+            "Y" => Self::fixed(-12),
+            "Z" => Self::utc(),
+            _ => {
+                lazy_static! {
+                    static ref HH_MM_MATCHER: Regex = Regex::new(r"^([\-\+]{1})(\d{2})(\d{2})$").unwrap();
```

Comment:
looking at the regex it looks like we only need to enable the `unicode-perl` and `std` features on `regex` in `spinoso-time`'s `Cargo.toml`.

Reply:
Just a quick thought - is including the whole regex package actually worth it for this one regex? I'm sure I could smash some rather ugly string manipulation code together for this same area, and it would be one less dependency :thinking:. I know artichoke-backend includes the `regex` crate, but it still feels a bit much for this piece of functionality

Reply:
yea there are some hand-rolled parsers in Artichoke already, like the one for `Kernel#Integer`:

https://github.com/artichoke/artichoke/blob/06e9e0d88dc7d7451dc5b2269e9d05c69bef5da6/artichoke-backend/src/extn/core/kernel/integer.rs#L328

For now this seems fine, but if you'd like to revisit this, let's file a ticket.

Reply:
I'm not super passionate about a non regex'd solution - but maybe one day for the fun if I remember instead of creating a ticket :smile: